### PR TITLE
rust 1.70.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

**Comment by chenrui333** [comment]
- relates to https://github.com/rust-lang/cargo/issues/12216

**Comment by carlocab** [comment]
Reran without build dependents. Original failures at https://github.com/Homebrew/homebrew-core/actions/runs/5164912796/attempts/2.

**Comment by chenrui333** [comment]
> Reran without build dependents. Original failures at https://github.com/Homebrew/homebrew-core/actions/runs/5164912796/attempts/2.

we might also need a rebase for some test fixes.

File: Formula/rust.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@ class Rust < Formula
   license any_of: ["Apache-2.0", "MIT"]
 
   stable do
-    url "https://static.rust-lang.org/dist/rustc-1.69.0-src.tar.gz"
-    sha256 "fb05971867ad6ccabbd3720279f5a94b99f61024923187b56bb5c455fa3cf60f"
+    url "https://static.rust-lang.org/dist/rustc-1.70.0-src.tar.gz"
+    sha256 "b2bfae000b7a5040e4ec4bbc50a09f21548190cb7570b0ed77358368413bd27c"
 
     # From https://github.com/rust-lang/rust/tree/#{version}/src/tools
     resource "cargo" do
```
Comment (carlocab) [inline Formula/rust.rb:11]: Doesn't this need to be updated?

Comment (chenrui333) [reply Formula/rust.rb:11]: oh shoot, my local branch does not have this change. My bad

Comment (chenrui333) [reply Formula/rust.rb:11]: Should be good now.

**Review by p-linnane — APPROVED** [review]

**Comment by chenrui333** [comment]
rebased all the linked rust 1.70 dependant PRs

and following up on the build dependency failures now (https://github.com/Homebrew/homebrew-core/actions/runs/5164912796/attempts/2)

**Comment by chenrui333** [comment]
- mathlibtools (regression build/test works fine in my local)
- watchman would be handled in #133417

**Comment by chenrui333** [comment]
- msc-generator can be tackled in the new version bump

**Comment by XVilka** [comment]
There is a problem with updated cargo for macOS - when it links system libcurl, it triggers the following bug:
- https://github.com/rust-lang/cargo/issues/12202
- https://github.com/rust-lang/cargo/pull/12234

The solution would be either pick up this patch, or link to the `brew`'s libcurl

**Comment by carlocab** [comment]
Let's pick up the patch then. Mind opening a PR, @XVilka? Thanks!

**Pull request merged**